e2e/scripts/wait-for-mrt-deploy.js

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+const fetch = require("node-fetch");
+const jsdom = require("jsdom");
+const program = require("commander");
+
+const DEFAULT_ORIGIN = process.env.CLOUD_API_BASE || "https://cloud.mobify.com";
+
+const doFetch = async (url, apiKey = null, method = "GET") => {
+  let response;
+  try {
+    response = await fetch(url, {
+      method: method,
+      headers: {
+        ...(apiKey && { Authorization: `Bearer ${apiKey}` }),
+      },
+    });
+  } catch (error) {
+    console.error(error);
+    process.exit(1);
+  }
+
+  if (!response.ok) {
+    console.error(
+      `${method} ${url}: Request failed with status ${response.status}`
+    );
+    process.exit(1);
+  }
+  return apiKey ? response.json() : response.text();
```

Comment:
Returning json vs text based on whether `apiKey` is set doesn't seem like an obvious behavior. Maybe add another param to the signature, `isJson = Boolean(apiKey)`?